resources/2023-06-resource-extensibility-product-spec.md

```diff
@@ -0,0 +1,133 @@
+# Resource Extensibility (User Defined Resource Types)
+
+## Summary
+The cloud native landscape is changing and expanding at a rapid pace like never before. Many enterprises use a wide range of technologies together for achieving their cloud-native strategy.  For any technology that’s a newcomer and helps solve a problem in the cloud native landscape, users look for an easy and seamless way to integrate their existing tools and technologies and incrementally adopt the new technology to their strategy. For Radius, we have heard requests from our customers/community to support technologies that their applications are tightly coupled with E.g.: an internal messaging service or a technology they absolutely love E.g.: PostgreSQL/Kafka.
+
+We need to enable the open-source community to build and experiment with imperfect things. We need to provide an extensibility model that supports “Bring your own technology”, define and use it with Radius. This will help us to meet the community where they are and enable them to experiment and leverage their work as open-source contributions.
+
+One of the high value extensibility points in Radius is Recipes. We have received interests to create custom resource types, define Recipes for the custom resource types and use it in the Radius application. Today Radius Extenders helps in creating custom resource types, but they are untyped and have limitations. The goal of providing resource extensibility is to empower developers or infrastructure operators to author and run their applications with custom resource types seamlessly in Radius and use all the other features such as Recipes, connections and app graph with ease and flexibility.
+
+### Top level goals
```

Comment:
I am still on the fence for merging User Defined Types and Recipes for any resource given how huge UDT area is. When I started writing the product specs, I felt these docs might be better on the level/abstraction of a roadmap item so that its easier for all stakeholders to comprehend and understand the design choices and nuances for a topic area. Open to other perspectives.